mackerel-plugin-linux/lib/linux.go

```diff
@@ -251,45 +252,61 @@ func parseProcStat(r io.Reader, p *map[string]interface{}) error {
 	return nil
 }
 
-// collect /proc/diskstats
-func collectProcDiskstats(path string, p *map[string]interface{}) error {
-	file, err := os.Open(path)
+// correct /sys/block/<device>/stat
+// See also. http://man7.org/linux/man-pages/man5/sysfs.5.html
+func collectDiskStats(path string, p *map[string]interface{}) error {
+	var elapsedData []mp.Metrics
+	var rwtimeData []mp.Metrics
+
+	sysBlockDir := filepath.Join(path, "block")
+
+	devices, err := ioutil.ReadDir(sysBlockDir)
 	if err != nil {
 		return err
 	}
-	defer file.Close()
-	return parseProcDiskstats(file, p)
-}
 
-// parsing metrics from diskstats
-func parseProcDiskstats(r io.Reader, p *map[string]interface{}) error {
-	var elapsedData []mp.Metrics
-	var rwtimeData []mp.Metrics
+	for _, d := range devices {
+		if d.Mode()&os.ModeSymlink != os.ModeSymlink {
```

Comment:
[question] what case is this condition true?

Reply:
There is no case that it becomes true, but it is checked just in case.

Reply:
I see